src/qibolab/platforms/abstract.py

```diff
@@ -15,6 +15,55 @@
 )
 
 
+class Qubit:
+    def __init__(self, name, settings, platform):
+        self.name = name
+        self.platform = platform
+        self.readout_frequency = settings[name]["resonator_freq"]
+        self.drive_frequency = settings[name]["qubit_freq"]
+        self.t1 = settings[name]["T1"]
+        self.t2 = settings[name]["T2"]
+        self.attenuation = None
+
+    def __getitem__(self):
+        return self.name
+
+    @property
+    def readout_frequency(self):
+        return self._readout_frequency
+
+    @readout_frequency.setter
+    def readout_frequency(self, frequency):
+        self._readout_frequency = frequency
+
+    @property
+    def drive_frequency(self):
+        return self._drive_frequency
+
+    @drive_frequency.setter
+    def drive_frequency(self, frequency):
+        self._drive_frequency = frequency
+
+    @property
+    def t1(self):
+        return self._t1
+
+    @t1.setter
+    def t1(self, time):
+        self._t1 = time
```

Comment:
If the property is simple and the setter doesn't do anything additional (eg. changing the platform) then you could also replace it with a single attribute `self.t1` defined in `__init__`.